### PR TITLE
Contrib extract

### PR DESCRIPTION
**Pull request by DavidDudson**
This is Phase 1 of 3.

I may want to beef up the test suite a little bit.
The only implicit added is `extract`. More may come at a later point in time.
This can be merged as is, and the others can be pushed separately

### CODE REVIEW

File: scalameta/contrib/src/main/scala/scala/meta/contrib/Extract.scala [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package scala.meta.contrib
+
+trait Extract[A, B] {
```
Comment (olafurpg) [inline scalameta/contrib/src/main/scala/scala/meta/contrib/Extract.scala]: Maybe add a docstring?

**Comment by olafurpg** [comment]
BTW. It seems scalafmt is negatively impacting productivity, is there something I can do to improve the workflow/make it go out of our way? Should we disable scalafmt? I'm totally fine with that too. cc/ @xavier-fernandez

**Comment by xavier-fernandez** [comment]
@olafurpg For me Scalafmt is ok, but I would enable it automatically the parsing check when doing `sbt test` on local machines, so it is hard to push unformatted code to the CI.

**Comment by olafurpg** [comment]
@xavier-fernandez I run tests all the time, and I don't want to slow that workflow down by reformatting all the time. I think a less intrusive approach is to put `./bin/scalafmt --diff` in `.git/hooks/pre-commit`

**Comment by DavidDudson** [comment]
Before I push a branch I run the below command

`git rebase --interactive --exec "./bin/scalafmt && git amend"`

It will rebase reformatting at every commit.

So you do not have to do it on commit, only on push. We could add a push hook.

**Comment by xavier-fernandez** [comment]
@DavidDudson I would rather have a push hook for avoiding problems :)

**Comment by DavidDudson** [comment]
@olafurpg The build is broken somehow https://platform-ci.scala-lang.org/scalameta/scalameta/1

**Comment by olafurpg** [comment]
@DavidDudson I am working on fixing the build to work on the new drone instance. Each iteration of setting up the CI is quite long because running sbt on a fresh docker instance is super slow. It would be really nice if sbt used coursier by default

**Comment by DavidDudson** [comment]
@olafurpg this may be of intrest https://github.com/sbt/sbt/issues/2997

**Comment by olafurpg** [comment]
The CI server should be fixed now, you may need to force push or create a new pull request to trigger the new webook.

**Comment by olafurpg** [comment]
It seems you will need to re-create a PR to use the new drone setup.

**Pull request closed without merging**